### PR TITLE
[fix]: [PL-18160]: Fixed wrong error msg for invalid branch name

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
Fixed wrong error msg for invalid branch name

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)

### CODE REVIEW

File: 952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java [hunk 1]
```diff
@@ -520,8 +521,10 @@ public CreateBranchResponse createNewBranch(
     } catch (WingsException e) {
       final WingsException cause = ExceptionUtils.cause(ErrorCode.SCM_UNPROCESSABLE_ENTITY, e);
       if (cause != null) {
-        throw new ExplanationException(
-            String.format("A branch with name %s already exists in the remote Git repository", branch), e);
+        throw new InvalidRequestException(String.format("Action could not be completed. Possible reasons can be:\n"
+                + "1. A branch with name %s already exists in the remote Git repository\n"
+                + "2. The branch name %s is invalid\n",
+            branch, branch));
       } else {
```
Comment (abhinav-harness) [inline 952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java:528]: Only problem i see it will be big for UI to display

**Review by bhavya181 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 526 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-nEg-33DYdPNYq5Pw:-Mx-nEg-33DYdPNYq5Px:bu9putd) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L526)):*
> ```Java
>         throw new InvalidRequestException(String.format("Action could not be completed. Possible reasons can be:\n"
>                 + "1. A branch with name %s already exists in the remote Git repository\n"
>                 + "2. The branch name %s is invalid\n",
> ```

Doesn't Git throw different exception for theses two cases, if yes then why we should use that, so that user can clearly get the right message. Also, we should throw exception here to get the stacktrace.

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 526 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-nEg-33DYdPNYq5Pw:-Mx-zQApDfmqKyXRqGuJ:bwfyqtt) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L526)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

Doesn't Git throw different exception for theses two cases, if yes then why we should use that, so that user can clearly get the right message. Also, we should throw exception here to get the stacktrace.
</blockquote></details>

github uses different terms which might not be easy to understand and discussed with @mohitg0795 we are not displaying error message thrown by github also it will further increase the size of message on ui

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 526 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-nEg-33DYdPNYq5Pw:-Mx01ZwZ8bZ_4e7dofs3:b-4pka3c) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L526)):*
<details><summary><i>Previously, rathodmeetsatish (Meet Rathod) wrote…</i></summary><blockquote>

github uses different terms which might not be easy to understand and discussed with @mohitg0795 we are not displaying error message thrown by github also it will further increase the size of message on ui
</blockquote></details>

@rathodmeetsatish please add log here for the incoming exception. @bhavya181 we see that if we attach the incoming exception and then wrap it with our exception, then everything gets displayed on UI, which displays Github error message containing irrelevant data. So, we try to avoid that use-case here.
___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 528 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-_4Ca8CQhxLu3Pznz-r1-528:-Mx01pOJCSbrTOoURS5V:bl6dxek) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L528)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

Only problem i see it will be big for UI to display
</blockquote></details>

I think that should be fine. We need to have good handling of UI messages going forward. Currently, I see much longer error messages on Pipeline use-cases in same wizard.

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 528 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-_4Ca8CQhxLu3Pznz-r1-528:-Mx3BP9L9J6a8euJnfkU:b-x7p2sn) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L528)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

I think that should be fine. We need to have good handling of UI messages going forward. Currently, I see much longer error messages on Pipeline use-cases in same wizard.
</blockquote></details>

@mohitg0795 agreed, but our problem of customers not knowing what to do will still be in place. IMO backend and frontend should see together if it fits in, if it doesn't someone should create a followup or fix it right away.

**Review by bhavya181 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 526 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-nEg-33DYdPNYq5Pw:-Mx3Qhiu3nfgUBzU3UwN:b-w0yspi) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L526)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

@rathodmeetsatish please add log here for the incoming exception. @bhavya181 we see that if we attach the incoming exception and then wrap it with our exception, then everything gets displayed on UI, which displays Github error message containing irrelevant data. So, we try to avoid that use-case here.
</blockquote></details>

got it.
___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 528 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-_4Ca8CQhxLu3Pznz-r1-528:-Mx3QmOhC2dcEKM9oDiA:bqc2jpp) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L528)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

@mohitg0795 agreed, but our problem of customers not knowing what to do will still be in place. IMO backend and frontend should see together if it fits in, if it doesn't someone should create a followup or fix it right away. 
</blockquote></details>

We already have a long message for bad credentials, if that is working, this will also fit on UI.

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 1 change requests. 3. unresolved discussions. 1 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 528 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-_4Ca8CQhxLu3Pznz-r1-528:-Mx3W8tc5c6lcNy6i7NG:b-ud7udo) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L528)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

We already have a long message for bad credentials, if that is working, this will also fit on UI.
</blockquote></details>

@bhavya181 can we check once, i highly doubt it. 
Also I have seen one more place where it doesn't where we give a doc link.

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 2 change requests. 3. unresolved discussions. 1 of 2 approvals obtained. 

___
*[952-scm-java-client/src/test/java/io/harness/impl/scm/ScmServiceClientImplTest.java, line 53 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx3YutQ95411tyg0Nn3:-Mx3YutQ95411tyg0Nn4:b-5nrzne) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/test/java/io/harness/impl/scm/ScmServiceClientImplTest.java#L53)):*
> ```Java
>     commit = Commit.newBuilder().setSha(sha).build();
>     getLatestCommitResponse = GetLatestCommitResponse.newBuilder().setCommit(commit).build();
>     createBranchResponse = CreateBranchResponse.newBuilder().setStatus(422).setError(error).build();
> ```

We don't need to pre-set these values. These can be just methods that can be used to generate different types of response objects and then directly used within test methods.



_Code quote:_
```Java
    commit = Commit.newBuilder().setSha(sha).build();
    getLatestCommitResponse = GetLatestCommitResponse.newBuilder().setCommit(commit).build();
    createBranchResponse = CreateBranchResponse.newBuilder().setStatus(422).setError(error).build();
  }
```

**Comment by rathodmeetsatish** [comment]
trigger messagecheck

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 526 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-nEg-33DYdPNYq5Pw:-Mx48UW3EdVDvPbtPejk:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L526)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

got it.
</blockquote></details>

Done.
___
*[952-scm-java-client/src/test/java/io/harness/impl/scm/ScmServiceClientImplTest.java, line 53 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx3YutQ95411tyg0Nn3:-Mx48YEs7SO0-Ubr8t_7:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/test/java/io/harness/impl/scm/ScmServiceClientImplTest.java#L53)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

We don't need to pre-set these values. These can be just methods that can be used to generate different types of response objects and then directly used within test methods.
</blockquote></details>

Done.

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 526 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-nEg-33DYdPNYq5Pw:-Mx4YRY399FQSNpYMF0o:bu7rrnz) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L526)):*
<details><summary><i>Previously, rathodmeetsatish (Meet Rathod) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

missing log line

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 526 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-nEg-33DYdPNYq5Pw:-Mx4xVZS2ffMHMK8hUxx:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L526)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

missing log line
</blockquote></details>

Done.
___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java, line 528 at r1](https://reviewable.io/reviews/harness/harness-core/31068#-Mx-_4Ca8CQhxLu3Pznz-r1-528:-Mx4xXSC5fkQpI5lVmr6:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/b267ff6a243b618461c26bf49c74328a373eef85/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmServiceClientImpl.java#L528)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

@bhavya181 can we check once, i highly doubt it. 
Also I have seen one more place where it doesn't where we give a doc link.
</blockquote></details>

Done.

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 1 of 3 files at r1, 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 3 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by mohitg0795 — DISMISSED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068#-:-Mx5fJ-7BipfIcBLcbhh:bnfp4nl)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by rijajoo — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068#-:-Mx82zr4-3VPSG5YTbvK:bnfp4nl)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by meenakshi-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: 1. unresolved discussions.

**Review by bhavya181 — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 1 of 1 files at r3, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31068)* status: :shipit: complete! ✓

**Pull request merged**